python/ql/src/semmle/python/dataflow/TaintTracking.qll

```diff
@@ -180,7 +180,10 @@ abstract class TaintKind extends string {
     TaintKind getTaintForIteration() { none() }
 
     predicate flowStep(DataFlow::Node fromnode, DataFlow::Node tonode, string edgeLabel) {
-        this.additionalFlowStepVar(fromnode.asVariable(), tonode.asVariable()) and
+        exists(DataFlowExtension::DataFlowVariable v |
+            v = fromnode.asVariable() and
+            v.getASuccessorVariable() = tonode.asVariable()
+        ) and
```

Comment:
I'm not immediately convinced that this part is safe:

1. Does this mean that `DataFlowExtension::DataFlowVariable` wasn't handled properly before :scream: 
2. If anyone was using `additionalFlowStepVar`, wouldn't this change make it stop work? I think either we should
    1. remove `additionalFlowStepVar` completely (if it has been deprecated long enough that it's safe to do so)
    2. keep supporting `additionalFlowStepVar` (in addition to `DataFlowExtension::DataFlowVariable`) until we can safely remove it.

Solution 2.ii probably means there might be _one_ deprecation warning, but right now I see that as the lesser evil

Reply:
> 1. Does this mean that `DataFlowExtension::DataFlowVariable` wasn't handled properly before :scream:

That would seem to be the conclusion. :slightly_frowning_face: 

> 2. If anyone was using `additionalFlowStepVar`, wouldn't this change make it stop work? I think either we should

Both versions of `additionalFlowStepVar` (the method on `TaintKind` and the one on `TaintFlow`) have been deprecated for a long time (years, as far as I can tell). No one should be using it.
 
Given how old this deprecation warning is, I think the best solution is 
>    i. remove `additionalFlowStepVar` completely (if it has been deprecated long enough that it's safe to do so)

(The problem with option ii being that we can never be sure if people are actually using this predicate, despite its deprecation, and so we would have to keep it around forever.)

Reply:
(To clarify, if we remove `additionalFlowStepVar` outright, then any `override`s will now get a warning, and so people may discover this breaking change that way.)